aten/src/ATen/native/TensorShape.cpp

```diff
@@ -1390,6 +1390,183 @@ Tensor index_select_sparse(const Tensor& self, int64_t dim, const Tensor& index)
   }
 }
 
+Tensor index_select_sparse_cpu(const Tensor& self, int64_t dim, const Tensor& index) {
+  /*
+    Algorithm:
+    index - a 1-D tensor of indicies with shape (n,)
+    self - sparse tensor, its shape is sizes = sparse_shape + dense_shape
+      indices - 2-D tensor of indices, shape is (sparse_dims, nnz)
```

Comment:
This is from the previous implementation. The current algorithm is a bit more involved, but not by much.